src/Chameleon/Chameleon.cs

```diff
@@ -116,93 +116,90 @@ private static void OnGetData(GetDataEventArgs args)
-                    var knownIps = JsonConvert.DeserializeObject<List<string>>(account.KnownIps);
-                    if (knownIps != null && player.IP == knownIps[^1])
-                    {
-                        if (player.State == 1)
-                        {
-                            player.State = 2;
-                        }
-
-                        NetMessage.SendData((int) PacketTypes.WorldInfo, player.Index);
-
-                        player.PlayerData = TShock.CharacterDB.GetPlayerData(player, account.ID);
-
-                        var group = TShock.Groups.GetGroupByName(account.Group);
-
-                        player.Group = group;
-                        player.tempGroup = null;
-                        player.Account = account;
-                        player.IsLoggedIn = true;
-                        player.IsDisabledForSSC = false;
-
-                        if (Main.ServerSideCharacter)
-                        {
-                            if (player.HasPermission(Permissions.bypassssc))
-                            {
-                                player.PlayerData.CopyCharacter(player);
-                                TShock.CharacterDB.InsertPlayerData(player);
-                            }
-                            player.PlayerData.RestoreCharacter(player);
-                        }
-                        player.LoginFailsBySsi = false;
-
-                        if (player.HasPermission(Permissions.ignorestackhackdetection))
-                        {
-                            player.IsDisabledForStackDetection = false;
-                        }
-
-                        if (player.HasPermission(Permissions.usebanneditem))
-                        {
-                            player.IsDisabledForBannedWearable = false;
-                        }
-
-                        player.SendSuccessMessage(GetString($"已经验证{account.Name}登录完毕。"));
-                        TShock.Log.ConsoleInfo(player.Name + GetString("成功验证登录。"));
-                        PlayerHooks.OnPlayerPostLogin(player);
-                        return true;
-                    }
-                    else
-                    {
-                        TShock.Log.ConsoleInfo(player.Name + GetString("IP地址不存在或与上次登录IP地址不符。"));
-                    }
-                }
-                else
+                TSPlayer.Server.SendMessage(GetString("DisableUUIDLogin被设置为true，无法通过UUID自动登录。"), Microsoft.Xna.Framework.Color.Yellow);
+                break;
+            }
+            if (account.UUID != player.UUID)
+            {
+                TShock.Log.ConsoleInfo(player.Name + GetString("登录UUID与上次不同。"));
+                break;
+            }
+            var knownIps = JsonConvert.DeserializeObject<List<string>>(account.KnownIps);
+            if (Configuration.Instance.VerifyloginIP && JsonConvert.DeserializeObject <string[]>(account.KnownIps) is var kownIps && (knownIps is null || player.IP != kownIps.LastOrDefault()))
```

Comment:
if (Configuration.Instance.VerifyloginIP && JsonConvert.DeserializeObject <string[]>(account.KnownIps) is var kownIps && (knownIps is null || player.IP != kownIps.Last()))